module/zfs/zfs_vnops.c

```diff
@@ -1050,6 +1050,140 @@ zfs_write(znode_t *zp, zfs_uio_t *uio, int ioflag, cred_t *cr)
 	return (0);
 }
 
+/*
+ * Rewrite a range of file as-is without modification.
+ *
+ *	IN:	zp	- znode of file to be rewritten.
+ *		off	- Offset of the range to rewrite.
+ *		len	- Length of the range to rewrite.
+ *		flags	- Random rewrite parameters.
+ *
+ *	RETURN:	0 if success
+ *		error code if failure
+ */
+int
+zfs_rewrite(znode_t *zp, uint64_t off, uint64_t len, uint64_t flags)
+{
+	int error;
+
+	(void) flags;
```

Comment:
Suggest reserving the flag space to known flags, ie:

```suggestion
	if (flags != 0)
	    return (SET_ERROR(EINVAL));
```

Reply:
I have no strong opinion about this.  All this function is quite advisory, so I am not sure we need to be strict here.  But if you'd prefer so, I'm fine with it.

Reply:
I prefer it as a general API design principle, just because I've seen people do weird things or make mistakes with args that got ignored and so seemed to work, and then later on came to life and broke something. This helps us avoid that from the outset.

Reply:
Done.